engine.py

```diff
@@ -175,6 +175,23 @@ def execute_command(self, cmd_key, args):
             # start up our QApp now, if none is already running
             qt_application = None
             if not QtGui.QApplication.instance():
+                if QtCore.qVersion()[0] == "5":
```

Comment:
can we rely on `self.has_qt5` [attribute](https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py#L626)?

Or maybe even move all your PR's updates to the Engine class and invoke that here (or when the object is instantiated)?

Reply:
I also first thought we could use the engine property `has_qt5`, but this does not guarantee that the `qt` base is Qt5, it only guarantees that there is a `qt5` base available and that base has an interface for Qt5 (so does not guarantee anything about the Qt version itself).

To simplify the version check, you could modify the `qt` base in tk-core engine.py to include a `version` attribute  that you could import like `from sgtk.platform.qt import version`. It's a little cleaner, but not sure it's worth the effort.